tests/e2e.py

```diff
@@ -0,0 +1,204 @@
+    self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+    self.assertTrue(created)
+    self.assertEqual(self.analysis_vm.name, 'new-vm-for-analysis')
+
+    if self.disk_to_forensic is not None:
+      # Make a copy of another disk of the instance to analyse
+      log.info('{0:s} disk copy started for instance: {1:s}.'
+               .format(self.disk_to_forensic, self.instance_to_analyse))
+      self.disk_to_forensic_copy = gcp.create_disk_copy(
+        src_proj=self.project_id,
+        dst_proj=self.project_id,
+        instance_name=self.instance_to_analyse,
+        zone=self.zone,
+        disk_name=self.disk_to_forensic
+      )
+      log.info('{0:s} disk successfully copied: {1:s}.'.
+               format(self.disk_to_forensic, self.disk_to_forensic_copy.name))
+      self.assertIsInstance(self.disk_to_forensic_copy, gcp.GoogleComputeDisk)
+      self.assertTrue(self.disk_to_forensic_copy.name.startswith('evidence-'))
+      self.assertTrue(self.disk_to_forensic_copy.name.endswith('-copy'))
+
+      # Use existing forensics VM and attach the other disk
+      log.info('Attaching disk {0:s} to existing instance {1:s}.'
+               .format(self.disk_to_forensic_copy.name, self.analysis_vm_name))
+      self.analysis_vm, created = gcp.start_analysis_vm(
+        project=self.project_id,
+        vm_name=self.analysis_vm_name,
+        zone=self.zone,
+        boot_disk_size=10,
+        cpu_cores=4,
+        attach_disk=self.disk_to_forensic_copy
+      )
+      self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+      self.assertFalse(created)
+      self.assertEqual(self.analysis_vm.name, 'new-vm-for-analysis')
+
+    disks = self.analysis_vm.list_disks()
+    log.warning(disks)
+    if self.disk_to_forensic is not None:
+      self.assertEqual(disks,
+                       ['new-vm-for-analysis',
+                        self.boot_disk_copy.name,
+                        self.disk_to_forensic_copy.name])
+    else:
+      self.assertEqual(disks,
+                       ['new-vm-for-analysis',
+                        self.boot_disk_copy.name])
+
+    # Cleanup after testing
+    self.__clean()
+
+  def __clean(self):
+    project = gcp.GoogleCloudProject(project_id=self.project_id, default_zone=self.zone)
+
+    # delete the created forensics VMs
+    log.info('Deleting analysis instance: {0:s}.'
+             .format(self.analysis_vm.name))
+    operation = project.gce_api().instances().delete(
+      project=project.project_id,
+      zone=self.zone,
+      instance=self.analysis_vm.name
+    ).execute()
+    try:
+      project.gce_operation(operation, block=True)
+    except HttpError:
+      pass
+    log.info('Instance {0:s} successfully deleted.'
+             .format(self.analysis_vm.name))
+
+    # We wait 5min before trying to delete the disk so that GCE propagates the previous operation,
+    # as the instance first needs to be stopped, and then deleted.
+    time.sleep(300)
```

Comment:
Ah, I remember discussing this. It's OK for now, but what would be /better/ would be to busy-loop around the existence or not of the VM that we're waiting to be deleted